pilot/pkg/bootstrap/validation.go

```diff
@@ -57,36 +43,11 @@ func (s *Server) initConfigValidation(args *PilotArgs) error {
 		return nil
 	})
 
-	if webhookConfigName := validationWebhookConfigName.Get(); webhookConfigName != "" && s.kubeClient != nil {
-		if webhookConfigName == validationWebhookConfigNameTemplate {
-			webhookConfigName = strings.ReplaceAll(validationWebhookConfigNameTemplate, validationWebhookConfigNameTemplateVar, args.Namespace)
-		}
-
-		o := controller.Options{
-			WatchedNamespace:  args.Namespace,
-			CABundleWatcher:   s.istiodCertBundleWatcher,
-			WebhookConfigName: webhookConfigName,
-			ServiceName:       "istiod",
-		}
-		s.addTerminatingStartFunc(func(stop <-chan struct{}) error {
-			leaderelection.
-				NewLeaderElection(args.Namespace, args.PodName, leaderelection.ValidationController, s.kubeClient).
-				AddRunFunction(func(leaderStop <-chan struct{}) {
-					whController, err := controller.New(o, s.kubeClient)
-					if err != nil {
-						log.Errorf("failed to start validation controller")
-						return
-					}
-					log.Infof("Starting validation controller")
-					// Start informers again. This fixes the case where informers for namespace do not start,
-					// as we create them only after acquiring the leader lock
-					// Note: stop here should be the overall pilot stop, NOT the leader election stop. We are
-					// basically lazy loading the informer, if we stop it when we lose the lock we will never
-					// recreate it again.
-					s.kubeClient.RunAndWait(stop)
-					whController.Start(leaderStop)
-				}).
-				Run(stop)
+	if s.kubeClient != nil {
```

Comment:
Since revisions patch their own webhooks don't need the leaderelection here anymore